### DISCUSSION TITLE
guidelines to add support to further devices

### DISCUSSION THREAD

**Discussion by rickespana** [syn-4ck/pynipper-ng#81] [General]
hello everyone, I was just wondering if there are any guidelines or docs to add support to further devices. I would like to contribute with adding support to more devices, starting with pfsense. Thank you.

**Comment by syn-4ck** [comment]
Hi @rickespana. 

Nowadays there aren't any guideline, but in this week I am going to write something to this.

In this moment, I am focus in Cisco and improving the tool architecture, but sounds good working together to add new devices.

Also, you can send me an email: repoJFM@protonmail.com. (Veo que eres de Colombia, si es así puedes escribirme en español).

Regards!

**Reply by rickespana** [reply]
Gracias! estaré pendiente

**Reply by syn-4ck** [reply]
Hola @rickespana! Acabo de subir unos cambios a la rama principal de `pynipper-ng`, pensados exclusivamente para facilitar la contribución y trabajo en equipo.

Además, he realizado cambios en el CONTRIBUTING.md, el fichero que actualmente mantiene las guias de como contribuir en esta herramienta. En estas guias encontrarás la información esencial para crear la estructura de un nuevo dispositivo (https://github.com/syn-4ck/pynipper-ng/blob/develop/CONTRIBUTING.md#new-device) y la implementación de los distintos plugins (https://github.com/syn-4ck/pynipper-ng/blob/develop/CONTRIBUTING.md#new-plugin). Esta guia está diseñada con el objetivo de explicar de forma sencilla y breve como crear una contribución, pero es buena idea ver el código de la herramienta para Cisco IOS y tomar ideas de ahí.

IMPORTANTE:  Recuerda seguir estos pasos antes de intentar incorporar los cambios (https://github.com/syn-4ck/pynipper-ng/blob/develop/CONTRIBUTING.md#getting-started). Dado que no se dan permisos de escritura en este repo, es necesario hacer un fork y usar unas buenas prácticas de desarrollo, creando una Pull Request del modo correcto y especificado en la guia y relacionandolo con una issue previamente creada.

Gracias por tu interés y espero ansioso tus contribuciones!

**Reply by rickespana** [reply]
muchas gracias por la información e interés, espero poder contribuir. :)
